test/OpenTelemetry.Tests/Metrics/AggregatorTestsBase.cs

```diff
@@ -238,6 +238,48 @@ public void MultiThreadedHistogramUpdateAndSnapShotTest()
         Assert.Equal(200, argsToThread.SumOfDelta + lastDelta);
     }
 
+    [Theory]
+    [InlineData("Microsoft.AspNetCore.Hosting", "http.server.request.duration")]
+    [InlineData("Microsoft.AspNetCore.Http.Connections", "signalr.server.connection.duration")]
+    [InlineData("Microsoft.AspNetCore.RateLimiting", "aspnetcore.rate_limiting.request_lease.duration")]
+    [InlineData("Microsoft.AspNetCore.RateLimiting", "aspnetcore.rate_limiting.request.time_in_queue")]
+    [InlineData("Microsoft.AspNetCore.Server.Kestrel", "kestrel.connection.duration")]
+    [InlineData("Microsoft.AspNetCore.Server.Kestrel", "kestrel.tls_handshake.duration")]
+    [InlineData("OpenTelemetry.Instrumentation.AspNetCore", "http.server.duration")]
+    [InlineData("OpenTelemetry.Instrumentation.Http", "http.client.duration")]
+    [InlineData("System.Net.Http", "http.client.connection.duration")]
+    [InlineData("System.Net.Http", "http.client.request.duration")]
+    [InlineData("System.Net.Http", "http.client.request.time_in_queue")]
+    [InlineData("System.Net.NameResolution", "dns.lookups.duration")]
+    public void HistogramBucketsDefaultUpdatesForSecondsTest(string meterName, string instrumentName)
+    {
+        RunTest(meterName, instrumentName, unit: "s");
+        RunTest(meterName, instrumentName, unit: "ms");
+        RunTest(meterName, instrumentName, unit: "By");
+        RunTest(meterName, instrumentName, unit: null);
+
+        void RunTest(string meterName, string instrumentName, string unit)
+        {
+            using var meter = new Meter(meterName);
+
+            var instrument = meter.CreateHistogram<double>(instrumentName, unit);
+
+            var metricStreamIdentity = new MetricStreamIdentity(instrument, metricStreamConfiguration: null);
+
+            AggregatorStore aggregatorStore = new(
```

Comment:
This way of testing is relying a bit too much on the knowledge of the internal implementation. Could you use test the bounds using an `InMemoryExporter` instead? Get hold of the `MetricPoint` after the export and check the bounds using `metricPoint.GetHistogramBuckets().ExplicitBounds`.

Reply:
I'm not familiar with InMemoryExporter, let me research it a bit. Thx

Reply:
> This way of testing is relying a bit too much on the knowledge of the internal implementation.

This logic is strange to me. It's a unit test. We're changing how `AggregatorStore` resolves buckets. That's the "unit" we are testing. Logic downstream reliant on the resolved buckets isn't being changed. Other tests _should_ be covering that. I would very much expect unit tests to cover internal details 😄 

If we want a bigger integration test exercising more of the SDK, fine to do, but too much of that is going to result in slow CI with a ton of overlap. Just my $0.02 😄

Reply:
> That's the "unit" we are testing.

We haven't really followed the pedantic approach to unit testing in our repo. It's a spectrum. The tests in our repo are always somewhere between a strict by-the-book unit test and a blanket integration test.

Reply:
Please make sure to add tests to cover that this special logic is only applicable if SDK is using defaults. i.e if a View is configured to override buckets, then that gets respected in all metrics, including the ones we are hardcoding.

Reply:
> Please make sure to add tests to cover that this special logic is only applicable if SDK is using defaults. i.e if a View is configured to override buckets, then that gets respected in all metrics, including the ones we are hardcoding.

And I recommend to the existing pattern of using InMemoryExporter to achieve this, though I fully agree its neither unit nor integration tests but something in between!

Reply:
> Please make sure to add tests to cover that this special logic is only applicable if SDK is using defaults. i.e if a View is configured to override buckets, then that gets respected in all metrics, including the ones we are hardcoding.

Why is that needed? This code is only touching the default bounds which are used if there wasn't a view. We have view tests covering view bounds are selected over the defaults (whatever they may be).

@samlam You could try extending [this test](https://github.com/open-telemetry/opentelemetry-dotnet/blob/34544bdb5effa80eca3facd7438763764a8063be/test/OpenTelemetry.Tests/Metrics/MetricViewTests.cs#L513). But not sure how to best solve needing an exact meter name.

> We haven't really followed the pedantic approach to unit testing in our repo. It's a spectrum. The tests in our repo are always somewhere between a strict by-the-book unit test and a blanket integration test.

Well we are the maintainers. If we ask people to add integration tests that's what we'll get 😄 And we'll have to maintain the extra code forever!

Reply:
> This code is only touching the default bounds which are used if there wasn't a view.

We know this to be true but it would be good to test that and verify that Views take precedence even for especially treated metrics.